enhancements/optional-manifest.md

```diff
@@ -0,0 +1,250 @@
+---
+title: optional-manifest
+authors:
+  - "@fgiloux"
+reviewers:
+  - "@njhale"
+approvers:
+  - TBD
+creation-date: 2021-10-06
+last-updated: 2021-10-06
+status: provisional
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# Optional Manifests
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [olm-docs](https://github.com/operator-framework/olm-docs/)
+
+## Summary
+
+Operator authors can ship additional resources with their operators by adding their manifest to the manifest directory inside the operator bundle. Today the creation of these resources is however mandatory for the InstallPlan, hence the installation of the operator, to succeed.
+This enhancement proposal is to allow for optional resources. When a resource is marked as optional the InstallPlan can carry on with the operator installation and eventually succeed even if the creation of the aforementioned resource failed.
+
+## Motivation
+
+Today when the API for the resource in the manifest directory does not exist in the cluster where the operator gets installed the InstallPlan fails.
+ServiceMonitor and VerticalPodAutoscaler are among the supported resources. They allow the operator authors to provide more advanced features (observability, autoscaling). These features are however not part of the core functionalities of most operators. As such the operator author may want to leverage them but not enforce them so that the operator bundle can be installed on clusters where they are not available. As these resources are not shipped with Kubernetes it is a significant drawback. It either holds operator authors of leveraging them or forces them to have additional logic directly in the operator for managing their creation.
+
+### Goals
+
+- Allow the specification of optional resources
+- Avoid InstallPlan failure when optional resources cannot be created due to the absence of their API
+- Keep the current behaviour for resources that are not specified as optional
+
+### Non-Goals
+
+- Although there is a relation between this enhancement proposal and optional dependencies it does not intend to cover the full scope of the later. It can however be seen as an enabler for it and aim at extending OLM in a way that is consumable by such an enhancement.
+
+## Proposal
+
+A resource provided in the manifest directory can be marked as optional with the following annotation:
```

Comment:
Can we illuminate the pros and cons of using an optional annotation vs having a property for the resource in the bundle metadata?